### PR TITLE
fix strings.split

### LINKED ISSUE #33623: tf.strings.split bug
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):binary
- TensorFlow version (use command below):1.15.0
- Python version:3.6.8
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
AttributeError: 'tensorflow.python.framework.ops.EagerTensor' object has no attribute 'to_sparse'
**Describe the expected behavior**
The op should return SparseTensor or RaggedTensor
**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
```python
import tensorflow as tf
tf.strings.split('a b')
```
**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

[colab](https://colab.research.google.com/drive/1PrFuL7hC25yRGmFfbwwRK1if9N8d1M9A)

### PR DESCRIPTION
**Pull request by fsx950223**
fix #33623

### CODE REVIEW

**Review by alextp — CHANGES_REQUESTED** [review]
Can I get a unit test which fails before and passes after this change?

**Comment by fsx950223** [comment]
> Can I get a unit test which fails before and passes after this change?

I will add unit test later

**Comment by fsx950223** [comment]
Test case added. @alextp.

**Comment by fsx950223** [comment]
According to document, strings_split_v1 should return ragged tensor or sparse tensor

**Review by alextp — DISMISSED** [review]

**Comment by rthadur** [comment]
@fsx950223 can you please fix the build failures ?

`third_party/tensorflow/python/framework/test_util.py", line 2892, in _assertRaggedEqual
    self.assertEqual(a_list, b_list, msg)
AssertionError: Lists differ:`

**Review by alextp — DISMISSED** [review]

**Comment by rthadur** [comment]
`AIL: Found 8 non-whitelisted pylint errors:
tensorflow/python/kernel_tests/string_split_op_test.py:282: [C0330(bad-continuation), ] Wrong continued indentation (remove 1 space).

tensorflow/python/kernel_tests/string_split_op_test.py:284: [C0330(bad-continuation), ] Wrong continued indentation (remove 1 space).

tensorflow/python/kernel_tests/string_split_op_test.py:286: [C0330(bad-continuation), ] Wrong continued indentation (remove 1 space).

tensorflow/python/kernel_tests/string_split_op_test.py:375: [C0330(bad-continuation), ] Wrong hanging indentation (add 4 spaces).

tensorflow/python/kernel_tests/string_split_op_test.py:390: [C0330(bad-continuation), ] Wrong continued indentation (remove 1 space).

tensorflow/python/kernel_tests/string_split_op_test.py:392: [C0330(bad-continuation), ] Wrong continued indentation (remove 1 space).

tensorflow/python/kernel_tests/string_split_op_test.py:394: [C0330(bad-continuation), ] Wrong continued indentation (remove 1 space).

tensorflow/python/kernel_tests/string_split_op_test.py:362: [W0622(redefined-builtin), StringSplitV2OpTest.testSplitV1] Redefining built-in 'input'`

@fsx950223  can you please fix this sanity errors ?

File: tensorflow/python/kernel_tests/string_split_op_test.py [hunk 1]
```diff
@@ -267,35 +267,130 @@ def testSplitV2(self,
...
+
+      {"testcase_name": "Ragged4DInputEmptySeparator",
+       "input": [[[[b"1 2 3", b"4"], [b"5 6"]], [[b"7 8 9"]]], [[[b""]]]],
+       "input_is_ragged": True,
+       "expected": [[[[[b"1", b"2", b"3"], [b"4"]], [[b"5", b"6"]]],
+                     [[[b"7", b"8", b"9"]]]], [[[[]]]]]},
+
+      ])  # pyformat: disable
+  def testSplitV1(self,
+                  input,
```
Comment (rthadur) [inline tensorflow/python/kernel_tests/string_split_op_test.py]: `FAIL: Found 1 non-whitelisted pylint errors:
tensorflow/python/kernel_tests/string_split_op_test.py:362: [W0622(redefined-builtin), StringSplitV2OpTest.testSplitV1] Redefining built-in 'input'



=== Sanity check step 3 of 15: do_pylint PYTHON3 (Python 3 pylint) ===`

**Review by rthadur — CHANGES_REQUESTED** [review]

File: tensorflow/python/kernel_tests/string_split_op_test.py (hunk 1 above)
Comment (rthadur) [reply tensorflow/python/kernel_tests/string_split_op_test.py]: can you please fix this sanity errors ? 
link: https://source.cloud.google.com/results/invocations/f0c921f7-3aa0-4705-aa15-661c770c53e0/log

Comment (fsx950223) [reply tensorflow/python/kernel_tests/string_split_op_test.py]: Changed.@rthadur

**Review by alextp — APPROVED** [review]

**Pull request merged**